patches/server/0982-Dont-resend-blocks-on-interactions.patch

```diff
@@ -0,0 +1,149 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Tue, 27 Jun 2023 21:09:11 -0400
+Subject: [PATCH] Dont resend blocks on interactions
+
+In general, the client now has an acknowledgment system which will prevent block changes made by the client to be reverted correctly.
+
+It should be noted that this system does not yet support block entities, so those still need to resynced when needed.
+
+diff --git a/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java b/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java
+index 2119c5baf31ba771d88c1569e22cf16631090b81..061ca9fbe4d6b56a056867190f7b00e0ab067e30 100644
+--- a/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java
++++ b/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java
+@@ -199,7 +199,7 @@ public class ServerPlayerGameMode {
+                 PlayerInteractEvent event = CraftEventFactory.callPlayerInteractEvent(this.player, Action.LEFT_CLICK_BLOCK, pos, direction, this.player.getInventory().getSelected(), InteractionHand.MAIN_HAND);
+                 if (event.isCancelled()) {
+                     // Let the client know the block still exists
+-                    this.player.connection.send(new ClientboundBlockUpdatePacket(this.level, pos));
++                    // Paper - Don't resync blocks
+                     // Update any tile entity data for this block
+                     capturedBlockEntity = true; // Paper - send block entity after predicting
+                     return;
+@@ -214,7 +214,7 @@ public class ServerPlayerGameMode {
+                 // Spigot start - handle debug stick left click for non-creative
+                 if (this.player.getMainHandItem().is(net.minecraft.world.item.Items.DEBUG_STICK)
+                         && ((net.minecraft.world.item.DebugStickItem) net.minecraft.world.item.Items.DEBUG_STICK).handleInteraction(this.player, this.level.getBlockState(pos), this.level, pos, false, this.player.getMainHandItem())) {
+-                    this.player.connection.send(new ClientboundBlockUpdatePacket(this.level, pos));
```

Comment:
If you're gonna add comments like this, please comment the line instead of drop + comment